### PR TITLE
Masking logic update + bug fixes

### PR DESCRIPTION
**Pull request by sarveshnagpal**
- Addressing user feedback to adjust default masking capabilities
- Updating unit tests to reflect new logic
- Updating mousewheel -> wheel event for better browser compatibility
- Bumping up version to 0.6.38

### CODE REVIEW

File: packages/clarity-js/types/data.d.ts [hunk 1]
```diff
@@ -247,7 +247,9 @@ export const enum Constant {
     UserId = "userId",
     SessionId = "sessionId",
     PageId = "pageId",
-    Mask = "•",
+    Mask = "•", // Placeholder character for explicitly masked content
+    Digit = "•", // Placeholder character for digits
```
Comment (toby-walker) [inline packages/clarity-js/types/data.d.ts:251]: >• [](http://example.com/codeflow?start=13&length=1)

Could we use a digit instead?

**Comment by toby-walker** [comment]
>         hasWhitespace = c === Data.Character.Tab || c === Data.Character.NewLine || c === Data.Character.Return || c === Data.Character.Blank;

Similarly this wouldn't handle other whitespace character classes -- e.g., Japanese might use a different character for a blank.

---
Refers to: packages/clarity-js/src/core/scrub.ts:86 in 9419b65. [](commit_id = 9419b65a1e6f3d5684cce96173e1d5cd836e79b7, deletion_comment = False)

**Review by toby-walker — APPROVED** [review]
:shipit:

File: packages/clarity-js/types/data.d.ts (hunk 1 above)
Comment (sarveshnagpal) [reply packages/clarity-js/types/data.d.ts:251]: I will tackle this in a follow on change. Currently keep it consistent with existing masking expectation.

**Comment by sarveshnagpal** [comment]
>         hasDigit = hasDigit || (c >= Data.Character.Zero && c <= Data.Character.Nine); // Check for digits in the current word

For now yes; since there are performance implications of moving regex outside this block. I will tackle that in a separate change.

---
In reply to: [1210040837](https://github.com/microsoft/clarity/pull/263#issuecomment-1210040837) [](http://example.com/codeflow?ancestors=1210040837,1210040837)

---
Refers to: packages/clarity-js/src/core/scrub.ts:84 in 9419b65. [](commit_id = 9419b65a1e6f3d5684cce96173e1d5cd836e79b7, deletion_comment = False)

**Comment by sarveshnagpal** [comment]
>         hasWhitespace = c === Data.Character.Tab || c === Data.Character.NewLine || c === Data.Character.Return || c === Data.Character.Blank;

Yes known but agreed to defer it to a different change to not regress performance.

---
In reply to: [1210041313](https://github.com/microsoft/clarity/pull/263#issuecomment-1210041313) [](http://example.com/codeflow?ancestors=1210041313,1210041313)

---
Refers to: packages/clarity-js/src/core/scrub.ts:86 in 9419b65. [](commit_id = 9419b65a1e6f3d5684cce96173e1d5cd836e79b7, deletion_comment = False)

File: packages/clarity-js/src/core/scrub.ts [hunk 2]
```diff
@@ -82,7 +90,18 @@ function redact(value: string): string {
             // Performance optimization: Lazy load string -> array conversion only when required
             if (hasDigit || hasEmail) {
                 if (array === null) { array = value.split(Data.Constant.Empty); }
-                mutate(array, spaceIndex, hasWhitespace ? i : i + 1);
+                // Work on a token at a time so we don't have to apply regex to a larger string
+                let token = value.substring(spaceIndex + 1, hasWhitespace ? i : i + 1);
+                // Check if unicode regex is supported, otherwise fallback to legacy regex
+                if (currencyRegex.unicode) {
```
[the remark this answers is not in the document]
Comment (sarveshnagpal) [reply packages/clarity-js/src/core/scrub.ts:96]: Technically redact and mask are different functions. Mask strips off every character while redact only looks for sensitive content through digits and email address. I will update the comment to make the distinction clearer.

**Review by ender336 — APPROVED** [review]

**Pull request merged**